webpack/assets/javascripts/react_app/components/Pagination/index.js

```diff
@@ -0,0 +1,159 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useHistory } from 'react-router-dom';
+import URI from 'urijs';
+import classNames from 'classnames';
+import {
+  Pagination as PF4Pagination,
+  PaginationVariant,
+} from '@patternfly/react-core';
+import { translate as __ } from '../../common/I18n';
+import { useForemanSettings } from '../../Root/Context/ForemanContext';
+import {
+  getURIpage,
+  getURIperPage,
+  changeQuery,
+} from '../../common/urlHelpers';
+import './index.scss';
+
+const Pagination = ({
+  itemCount,
+  onSetPage,
+  onPerPageSelect,
+  onChange,
+  className,
+  page: propsPage,
+  perPage: propsPerPage,
+  noSidePadding,
+  variant,
+  updateParamsByUrl,
+  ...props
+}) => {
+  const { perPage: settingsPerPage = 20 } = useForemanSettings() || {};
+  const [page, setPage] = useState(propsPage);
+  const [perPage, setPerPage] = useState(propsPerPage || settingsPerPage);
+  const history = useHistory();
+  const { location: { key: locationKey, search } = {} } = history || {};
+
+  useEffect(() => {
+    let nextPage = propsPage;
+    let nextPerPage = propsPerPage;
+    if (updateParamsByUrl) {
+      if (locationKey) {
```

Comment:
Found the reason for this regression :)
There is no `key` prop in location, can't we just check if `search` exists instead?

Reply:
Done :)